numpy/lib/shape_base.py

```diff
@@ -1155,20 +1156,20 @@ def kron(a, b):
         b = reshape(b, bs)
 
     # Equalise the shapes by prepending smaller one with 1s
-    as_ = (1,)*max(0, ndb-nda) + as_
-    bs = (1,)*max(0, nda-ndb) + bs
+    a_extend_dims, b_extend_dims = (1,)*max(0, ndb-nda), (1,)*max(0, nda-ndb)
+    as_, bs = a_extend_dims + as_, b_extend_dims + bs
```

Comment:
Why did you make this change?

Reply:
Oh my bad, I was trying something where I was using with `a_extend_dims` but then stashed that idea. I'll put it back.

Reply:
Removed and rebased via [968ca6c](https://github.com/numpy/numpy/pull/21354/commits/968ca6caf165ed846b5e9448294a095dcd6c3d15)